src/webserver/DOSGuard.h

```diff
@@ -170,11 +190,49 @@ class BasicDOSGuard : public BaseDOSGuard
 
         {
             std::unique_lock lck(mtx_);
-            auto it = ipFetchCount_.find(ip);
-            if (it == ipFetchCount_.end())
-                ipFetchCount_[ip] = numObjects;
+            auto it = ipState_.find(ip);
+            if (it == ipState_.end())
+            {
+                ipState_.emplace(ip, ClientState{.TransferedByte = numObjects});
```

Comment:
I have a suspicion that this code will not compile on clang. Parenthesized initialization was not supported a month ago so chances are it still is not.. I'm not sure about AppleClang but i guess same story?

Reply:
Also, can we not rely on `operator[]` default constructing a zeroed ClinetState here?
So like simplify all this to `ipState_[ip].bytesTransferred += numObjects;`. Unless i'm misunderstanding something. WDYT?

Reply:
Great comment. fix it.